examples/asr/emformer_rnnt/train.py

```diff
@@ -29,65 +31,68 @@ def run_train(args):
         checkpoint,
         train_checkpoint,
     ]
-    trainer = Trainer(
+    return Trainer(
         default_root_dir=args.exp_dir,
         max_epochs=args.epochs,
         num_nodes=args.num_nodes,
         gpus=args.gpus,
         accelerator="gpu",
         strategy="ddp",
-        gradient_clip_val=5.0,
+        gradient_clip_val=args.gradient_clip_val,
         callbacks=callbacks,
     )
 
-    model = RNNTModule(
-        tedlium_path=str(args.tedlium_path),
-        sp_model_path=str(args.sp_model_path),
-        global_stats_path=str(args.global_stats_path),
-        reduction=args.reduction,
-    )
-    trainer.fit(model)
 
+def get_lightning_module(args):
+    if args.model_type == MODEL_TYPE_LIBRISPEECH:
+        return LibriSpeechRNNTModule(
+            librispeech_path=str(args.dataset_path),
+            sp_model_path=str(args.sp_model_path),
+            global_stats_path=str(args.global_stats_path),
+        )
+    elif args.model_type == MODEL_TYPE_TEDLIUM3:
+        return TEDLIUM3RNNTModule(
+            tedlium_path=str(args.dataset_path),
+            sp_model_path=str(args.sp_model_path),
+            global_stats_path=str(args.global_stats_path),
+        )
+    else:
+        raise ValueError(f"Encountered unsupported model type {args.model_type}.")
 
-def _parse_args():
-    parser = ArgumentParser(
-        description=__doc__,
-        formatter_class=RawTextHelpFormatter,
-    )
-    parser.add_argument(
-        "--exp-dir",
-        default=pathlib.Path("./exp"),
-        type=pathlib.Path,
-        help="Directory to save checkpoints and logs to. (Default: './exp')",
-    )
+
+def parse_args():
+    parser = ArgumentParser()
+    parser.add_argument("--model_type", type=str, choices=[MODEL_TYPE_LIBRISPEECH, MODEL_TYPE_TEDLIUM3], required=True)
     parser.add_argument(
-        "--global-stats-path",
+        "--global_stats_path",
         default=pathlib.Path("global_stats.json"),
         type=pathlib.Path,
         help="Path to JSON file containing feature means and stddevs.",
+        required=True,
```

Comment:
As it has a default value, is it really necessary to specify `required=True` here?
Also, the default value is not correct as that file is inside the recipe dir.

I would suggest removing the default value.